test/rules/strict-boolean-expressions/allow-string/test.ts.lint

```diff
@@ -0,0 +1,14 @@
+declare function get<T>(): T;
+
+if (get<string>()) {}
+
+if (get<boolean | string>()) {}
+    ~~~~~~~~~~~~~~~~~~~~~~~ [If statement condition mixes multiple kinds of falsy values. Expected a boolean or string.]
```

Comment:
Changed "one or more" to "more than one" since it allows having exactly one falsy type.